scripts/create_github_releases/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// tag must specify major.minor.patch or major.minor.patch-rc#
+	// (e.g. 1.12.1, 1.12.1-rc1)
+	validTag = regexp.MustCompile(`^\d+\.\d+\.\d+(-rc\d+)?$`)
+	// target must specify release branch for releasing
+	// or main for release candidates
+	validTarget = regexp.MustCompile(`^(release-\d+-\d+|main)$`)
+	// true or false
+	validPrerelease = regexp.MustCompile(`^(true|false)$`)
+	// yes or no
+	validYesNo = regexp.MustCompile(`^(yes|no)$`)
+)
+
+func scanInputFromStdio(policy *regexp.Regexp, format *string, question, errMsg string) error {
+	fmt.Println(question)
+	fmt.Print("> ")
+	fmt.Scanf("%s", format)
+
+	// validate policy
+	if !policy.Match([]byte(*format)) {
+		return errors.New(errMsg)
+	}
+	return nil
+}
+
+// validateRelease validates policies for releases
+// used for release candidates. Return target branch name
+// based on the specified tag name.
+func validateRelease(tag, preRelease string) (string, error) {
+	if strings.Contains(tag, "-rc") {
+		// release candidates must be released as prereleases
+		if preRelease == "false" {
+			return "", errors.New("[ERROR] Prerelease must be true for release candidates")
+		}
+		// release candidates must use main branch as a target
+		return "main", nil
+	} else {
+		versions := strings.Split(tag, ".")
+		// naming convention for release branch is release-{major}-{minor}
+		return fmt.Sprintf("release-%s-%s", versions[0], versions[1]), nil
+	}
+}
+
+func main() {
+	fmt.Println("Start creating new GitHub Release interactively!")
+
+	var tag, target, preRelease, confirm string
+
+	if err := scanInputFromStdio(
+		validTag,
+		&tag,
+		"Q. Input version tag with the format {major}.{minor}.{patch} or {major}.{minor}.{patch}-{rc#} (e.g. 1.19.0 or 1.19.0-rc1)",
+		"[ERROR] Version tag must follow the format {major}.{minor}.{patch} or {major}.{minor}.{patch}-{rc#}",
+	); err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	if err := scanInputFromStdio(
+		validPrerelease,
+		&preRelease,
+		"Q. Is prerelease? (Set true for release candidates) (e.g. true or false)",
+		"[ERROR] Prerelease must be true or false",
+	); err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	target, err := validateRelease(tag, preRelease)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	title := fmt.Sprintf("Release %s", tag)
+	// Encode white space with percent encoding
+	// because url.QueryEscape does not support
+	// percent encoding.
+	title = strings.Replace(title, " ", "%20", -1)
+
+	repositoryURL := "https://github.com/cookpad/terraform-aws-eks"
```

Comment:
Probably should be a const (if nothing else, to stop it being buried this far down into the code).

Reply:
♻️ https://github.com/cookpad/terraform-aws-eks/pull/264/commits/7f98d7fb73098170c7b5bf821250978f80cca3f9